x-pack/plugins/fleet/public/applications/fleet/sections/agents/agent_list_page/components/tags.tsx

```diff
@@ -13,21 +13,35 @@ interface Props {
   tags: string[];
 }
 
+// Number of tags displayed before "+ N more" is displayed
 const MAX_TAGS_TO_DISPLAY = 3;
 
+// Number of characters to display for each tag before truncation
+const MAX_TAG_DISPLAY_LENGTH = 20;
+
+function truncateTag(tag: string) {
+  return tag.length > MAX_TAG_DISPLAY_LENGTH
+    ? `${tag.substring(0, MAX_TAG_DISPLAY_LENGTH)}...`
+    : tag;
+}
+
 export const Tags: React.FunctionComponent<Props> = ({ tags }) => {
   return (
     <>
       {tags.length > MAX_TAGS_TO_DISPLAY ? (
         <>
-          <EuiToolTip content={<span data-test-subj="agentTagsTooltip">{tags.join(', ')}</span>}>
+          <EuiToolTip
+            content={
+              <span data-test-subj="agentTagsTooltip">{tags.map(truncateTag).join(', ')}</span>
+            }
+          >
             <span data-test-subj="agentTags">
-              {take(tags, 3).join(', ')} + {tags.length - MAX_TAGS_TO_DISPLAY} more
+              {take(tags, 3).map(truncateTag).join(', ')} + {tags.length - MAX_TAGS_TO_DISPLAY} more
             </span>
           </EuiToolTip>
         </>
       ) : (
-        <span data-test-subj="agentTags">{tags.join(', ')}</span>
+        <span data-test-subj="agentTags">{tags.map(truncateTag).join(', ')}</span>
```

Comment:
Should we show this in a tooltip if we truncate tag?

Reply:
Yes we can do this in the truncates tags that render in the column, but if a truncated tag renders inside another tooltip I don't think we can do much. 

Thinking about it more, I don't know that we should truncate tags in the tooltips. Maybe just set a max-width so they wrap. See the example in the original issue for what I mean, where the tooltip becomes very wide.

Reply:
Addressed in https://github.com/elastic/kibana/commit/6bdb7fcd5e4b36a2df8756ebe6e0ef1b5b1e8108, see comment below